examples/perplexity/perplexity.cpp

```diff
@@ -500,12 +499,16 @@ static results_perplexity perplexity(llama_context * ctx, const gpt_params & par
 
     const int num_batches = (n_ctx + n_batch - 1) / n_batch;
 
+    GGML_ASSERT(n_batch < n_ctx || n_batch % n_ctx == 0);
+    const int n_seq = std::max(1, n_batch / n_ctx);
```

Comment:
The uppermost limit for `seq_id`s should come from the newly added `llama_n_max_seq()`, or else this won't work properly for Mamba (see <https://github.com/ggerganov/llama.cpp/pull/5328#issuecomment-1985953811>). I can further explain if required.

Also, since `n_batch` is assumed to be smaller than `n_ctx`, it seems the division is reversed and would always result in 0.

I did not test this suggestion, but hopefully it helps.
```suggestion
    const int n_seq = std::min((int) llama_n_max_seq(ctx), std::max(1, n_ctx / n_batch));
```

Reply:
The division is correct, the goal is to find how many sequences at full context we can use at the same time, given the batch size. There are two cases supported, either the batch size is smaller than the context, or it is a multiple of the context size. This is actually enforced in `main`, so this assert should never be hit. I have pushed a change to set `n_parallel` to `n_seq` during context creation.

Reply:
Oh, right I had forgotten what `n_ctx` meant specifically when calculating perplexity. In this case, yes, setting `n_parallel` to `n_seq` is the right thing to do.